pkg/azuredx/adxauth/adxusercontext/from_req.go

```diff
@@ -0,0 +1,98 @@
+package adxusercontext
+
+import (
+	"context"
+	"strings"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+func FromQueryReq(ctx context.Context, req *backend.QueryDataRequest) context.Context {
+	if req == nil {
+		return ctx
+	}
+
+	idToken := getQueryReqHeader(req, "X-ID-Token")
+	accessToken := extractBearerToken(getQueryReqHeader(req, "Authorization"))
+
+	currentUser := CurrentUserContext{
+		User:        req.PluginContext.User,
+		IdToken:     idToken,
+		AccessToken: accessToken,
+	}
+
+	return WithCurrentUser(ctx, currentUser)
+}
+
+func FromResourceReq(ctx context.Context, req *backend.CallResourceRequest) context.Context {
+	if req == nil {
+		return ctx
+	}
+
+	idToken := getResourceReqHeader(req, "X-ID-Token")
+	accessToken := extractBearerToken(getResourceReqHeader(req, "Authorization"))
+
+	currentUser := CurrentUserContext{
+		User:        req.PluginContext.User,
+		IdToken:     idToken,
+		AccessToken: accessToken,
+	}
+
+	return WithCurrentUser(ctx, currentUser)
+}
+
+func FromHealthCheckReq(ctx context.Context, req *backend.CheckHealthRequest) context.Context {
+	if req == nil {
+		return ctx
+	}
+
+	// TODO: Add support for ID and access tokens
```

Comment:
can you add a link here in the comment to the issue that is blocking this TODO?

The same applies to the other TODO comments that you have added. A link to an issue helps to know when the TODO can be resolved in the future.

Reply:
Yes, will do